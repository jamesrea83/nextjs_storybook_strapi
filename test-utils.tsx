import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';

import { Themes } from './styles/themes';

interface Props {
	children: ReactNode;
}

const Wrapper = ({ children }: Props) => (
	<ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
	render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react';

export { customRender as render };
