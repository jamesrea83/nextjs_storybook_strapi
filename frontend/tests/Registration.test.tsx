import { render, screen, act } from '@/test-utils';
import userEvent from '@testing-library/user-event';

import Registration from '@/pages/registration';

describe('Registration page', () => {
	it('should render', () => {
		const { container } = render(<Registration />);
		expect(container).toMatchSnapshot();
	});

	it('should validate input', async () => {
		render(<Registration />);

		const usernameInput = screen.getByRole('textbox', { name: 'Username' });
		const emailInput = screen.getByRole('textbox', { name: 'Email' });
		const passwordInput = screen.getByRole('textbox', { name: 'Password' });
		const submitButton = screen.getByRole('button', { name: 'Sign up' });

		await act(() => userEvent.click(submitButton));

		expect(screen.getAllByText('Required field')).toHaveLength(3);

		await act(() => userEvent.type(usernameInput, 'test'));

		expect(screen.getByText('Min length 6!')).toBeInTheDocument();
		expect(screen.getAllByText('Required field')).toHaveLength(2);

		await act(() => userEvent.type(passwordInput, 'test'));

		expect(screen.getAllByText('Min length 6!')).toHaveLength(2);

		await act(() => userEvent.type(emailInput, 'test'));

		expect(screen.getByText('Invalid email')).toBeInTheDocument();

		await act(() => userEvent.type(usernameInput, 'testtest'));
		await act(() => userEvent.type(passwordInput, 'testtesttest'));
		await act(() => userEvent.type(emailInput, 'test@test.com'));

		expect(screen.queryByText('Min length 6!')).not.toBeInTheDocument();
		expect(screen.queryByText('Invalid email')).not.toBeInTheDocument();
	});
});
