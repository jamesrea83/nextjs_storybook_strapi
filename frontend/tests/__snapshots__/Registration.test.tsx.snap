// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration page should render 1`] = `
<div>
  <form>
    <div
      class="css-42099h"
    >
      <section
        class="css-1beoq7g"
      >
        <h2>
          Create an account
        </h2>
        <h3>
           
        </h3>
        <label
          class="css-1adzpiz"
          height="7"
          width="20"
        >
          <span
            class="css-porbx7"
          >
            Username
          </span>
          <div
            class="css-bnug8b"
          >
            <input
              class="css-58wzc"
              name="username"
              placeholder="username"
            />
          </div>
          <span
            class="css-tx5ech"
          >
             
          </span>
        </label>
        <label
          class="css-1adzpiz"
          height="7"
          width="20"
        >
          <span
            class="css-porbx7"
          >
            Email
          </span>
          <div
            class="css-bnug8b"
          >
            <input
              class="css-58wzc"
              name="email"
              placeholder="email"
            />
          </div>
          <span
            class="css-tx5ech"
          >
             
          </span>
        </label>
        <label
          class="css-1adzpiz"
          height="7"
          width="20"
        >
          <span
            class="css-porbx7"
          >
            Password
          </span>
          <div
            class="css-bnug8b"
          >
            <input
              class="css-58wzc"
              name="password"
              placeholder="password"
            />
          </div>
          <span
            class="css-tx5ech"
          >
             
          </span>
        </label>
        <button
          class="css-odwgcq"
          color="primary"
          type="submit"
        >
          Sign up
        </button>
        <h3>
          <a
            class="css-yf649i"
            href="/login"
            underline="1"
          >
            Log in
          </a>
        </h3>
      </section>
    </div>
  </form>
</div>
`;
